### PR TITLE
fix(#2535): TextYankPost event sends vim.v.event

### PR DESCRIPTION
**Pull request by zyd2001**
Try to fix issue https://github.com/nvim-tree/nvim-tree.lua/issues/2535#issuecomment-2028457412 that I also encountered.

I write the data to a temporary buffer and use vim 'y' command to do real yank to trigger TextYankPost event with correct v:event.

### CODE REVIEW

File: lua/nvim-tree/actions/fs/copy-paste.lua [hunk 1]
```diff
@@ -276,17 +276,22 @@ end
...
+    reg = '+'
   else
-    vim.fn.setreg('"', content)
-    vim.fn.setreg("1", content)
     clipboard_name = "neovim"
+    reg = '1'
   end
 
-  vim.api.nvim_exec_autocmds("TextYankPost", {})
+  local temp_buf = vim.api.nvim_create_buf(false, true)
```
Comment (alex-courtis) [inline lua/nvim-tree/actions/fs/copy-paste.lua:288]:
```suggestion
  -- manually firing TextYankPost does not set vim.v.event
  -- workaround: create a scratch buffer with the clipboard contents and send a yank command
  local temp_buf = vim.api.nvim_create_buf(false, true)
```

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Looking good, although it fails on `gy` and friends:

```
E5108: Error executing lua: ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:290: Error executing lua: ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:291: bad argument #2 to 'format' (number expected, got string)
stack traceback:
        [C]: in function 'format'
        ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:291: in function <...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:290>
        [C]: in function 'nvim_buf_call'
        ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:290: in function 'copy_to_clipboard'
        ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:300: in function 'fn'
        .../nd/site/pack/packer/start/zyd2001/lua/nvim-tree/api.lua:65: in function <.../nd/site/pack/packer/start/zyd2001/lua/nvim-tree/api.lua:62>
stack traceback:
        [C]: in function 'nvim_buf_call'
        ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:290: in function 'copy_to_clipboard'
        ...er/start/zyd2001/lua/nvim-tree/actions/fs/copy-paste.lua:300: in function 'fn'
        .../nd/site/pack/packer/start/zyd2001/lua/nvim-tree/api.lua:65: in function <.../nd/site/pack/packer/start/zyd2001/lua/nvim-tree/api.lua:62>
```

Clean room with 
```lua
vim.api.nvim_create_autocmd("TextYankPost", {
  callback = function(data)
    log.line("dev", "TextYankPost\ndata=%s\nevent=%s", vim.inspect(data), vim.inspect(vim.v.event))
  end,
})
```

**Comment by zyd2001** [comment]
Should be working now, used wrong format string.

**Review by alex-courtis — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**